_scripts/add_concierge_discussions/main.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/nedrocks/delphisbe/graph/model"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/endpointcreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/nedrocks/delphisbe/internal/backend"
+	"github.com/nedrocks/delphisbe/internal/config"
+	"github.com/nedrocks/delphisbe/internal/secrets"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultPort = "8080"
+)
+
+func main() {
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.Debugf("Starting")
+
+	ctx := context.Background()
+	rand.Seed(time.Now().Unix())
+
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = defaultPort
+	}
+
+	config.AddConfigDirectory("./config")
+	config.AddConfigDirectory("/var/delphis/config")
+	conf, err := config.ReadConfig()
+	if err != nil {
+		logrus.WithError(err).Errorf("Error loading config file")
+		return
+	}
+	logrus.Debugf("Got config from file")
+
+	awsConfig := aws.NewConfig().WithRegion(conf.AWS.Region).WithCredentialsChainVerboseErrors(true)
+	var awsSession *session.Session
+	if conf.AWS.UseCredentials {
+		awsConfig = awsConfig.WithCredentials(credentials.NewStaticCredentials(
+			conf.AWS.Credentials.ID, conf.AWS.Credentials.Secret, conf.AWS.Credentials.Token))
+	} else if conf.AWS.IsFargate {
+		if ECSCredentialsURI, exists := os.LookupEnv("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI"); exists {
+			endpoint := fmt.Sprintf("http://169.254.170.2%s", ECSCredentialsURI)
+			awsSession = session.New(awsConfig)
+			providerClient := endpointcreds.NewProviderClient(*awsSession.Config, awsSession.Handlers, endpoint)
+			creds := credentials.NewCredentials(providerClient)
+			awsConfig = awsConfig.WithCredentials(creds)
+		}
+	}
+	logrus.Debugf("Got creds from remote")
+	awsSession = session.Must(session.NewSession(awsConfig))
+
+	secretManager := secrets.NewSecretsManager(awsConfig, awsSession)
+	secrets, err := secretManager.GetSecrets()
+	logrus.Debugf("Got secrets")
+	if err == nil {
+		for k, v := range secrets {
+			os.Setenv(k, v)
+		}
+		conf.ReadEnvAndUpdate()
+	}
+
+	logrus.Debugf("about to create backend")
+	delphisBackend := backend.NewDelphisBackend(*conf, awsSession)
+	logrus.Debugf("Created backend")
+
+	doWork(ctx, delphisBackend)
+
+}
+
+func doWork(ctx context.Context, delphisBackend backend.DelphisBackend) {
+	// Fetch discussions
+	connection, err := delphisBackend.ListDiscussions(ctx)
```

Comment:
We should be slightly careful here because we need to update ListDiscussions to only show the available discussions.

I _think_ this script is only a 1-time backfill thing so it's probably not important, but if we run it again in the future we could end up in a strange state where we only get some discussions here.

Reply:
makes sense. I imagine we'll have to tweak this script if we ever have to run it again